corehq/messaging/scheduling/scheduling_partitioned/dbaccessors.py

```diff
@@ -157,6 +183,24 @@ def get_case_timed_schedule_instances_for_schedule(case_id, schedule):
     return get_case_timed_schedule_instances_for_schedule_id(case_id, schedule.schedule_id)
 
 
+def get_case_schedule_instance(cls, case_id, schedule_instance_id):
+    from corehq.messaging.scheduling.scheduling_partitioned.models import (
+        CaseAlertScheduleInstance,
+        CaseTimedScheduleInstance,
+    )
+
+    if cls not in (CaseAlertScheduleInstance, CaseTimedScheduleInstance):
+        raise ValueError("Expected CaseAlertScheduleInstance or CaseTimedScheduleInstance")
```

Comment:
Is it intentional that `get_active_schedule_instance_ids ` raises `TypeError`, but this one raises `ValuError`. Might be worth to make all of these a custom exception, so that it can be handled more surgically